### PR TITLE
Using BeautifulSoup, which is more flexible and runs on macOS 11 beta.

### PR DESCRIPTION
**Pull request by AndiCui**
Please test the Pipfile, I cannot get it to work on my system.

### CODE REVIEW

**Comment by AndiCui** [comment]
I think there is a couple of things that can be done -- like checking all products. Maybe with an algorithm to determine whether if the deal is worth it (enough percentage?) etc.

**Comment by zmoog** [comment]
Hey @AndiCui, thank you for your contribution! I totally missed this PR, I'll check it out but it looks great, thank you! ❤️

File: refurbished/parser.py [hunk 1]
```diff
@@ -3,74 +3,106 @@
...
-    return f'https://www.apple.com{href}'
+def _parse_url(product: bs4.element.Tag, store_domain: str) -> str:
+    """
+    Parse the fragment with product URL.
+    Just appending the relative path onto apple.com does not work for all regions.
+    For example, apple.com/cn redirects to apple.com.cn.
+    All links are then relative to the .cn domain.
+    """
+    href = product.h3.a.attrs["href"]
+    return store_domain + href
```
Comment (zmoog) [inline refurbished/parser.py:100]: I didn't know this edge case, now it's handled nicely.

**Review by zmoog — APPROVED** [review]
LGTM, a lot of improvement, thank you for your contribution!

**Pull request merged**